Need a better way to rebuild submodules

How should we update our Makefile to make submodules easier to manage?

Now that we are updating our submodules on every release it is becoming cumbersome to frequently hit this error from the Makefile:

```
$ make
SUBMODULES
Error: Submodule version mismatch
luajit:
  require: v2.0.4-306-gfe56522
  found:   v2.0.4-306-gfe56522
ljsyscall:
  require: v0.10-65-g7081d97
  found:   v0.10-65-g7081d97 
pflua:
  require: 5e2c56baa0cf1ec471719bac83e2a99c4e2d5495
  found:   27523953c03745d4c4fbbcf81f676bc774e4329e 
Please update and rebuild submodules.

Cheat sheet:
  cd snabbswitch
  git submodule update
  make clean
  make -j

make: *** [snabb] Error 1
```

This was added because previously we had the much worse problem of accidentally running with the wrong submodule versions. This error check was added when we found a LuaJIT fix that we absolutely depended on being present.

I suppose we have a couple of options:

1. We could automate those steps. The issue here is that I don't really like having `make` in one directory recurse *upwards* to the parent directory and I wonder if this will cause confusion.
2. We could add a new target in `src/Makefile` that does these steps. So then you would write `make submods` instead of `(cd ..; git submodule update; make clean; make)`.

--- a/src/Makefile
+++ b/src/Makefile
@@ -71,13 +71,9 @@ snabb: $(LUAOBJ) $(PFLUAOBJ) $(HOBJ) $(COBJ) $(ARCHOBJ) $(ASMOBJ) $(INCOBJ) $(LU
 	    echo "pflua:"						>&2; \
 	    echo "  require: $(PFLUA_VSN)"				>&2; \
 	    echo "  found:   $(shell cat ../deps/pflua.vsn) "		>&2; \
-	    echo "Please update and rebuild submodules."		>&2; \
-	    echo ""							>&2; \
-	    echo "Cheat sheet:"						>&2; \
-	    echo "  cd snabbswitch"					>&2; \
-	    echo "  git submodule update"				>&2; \
-	    echo "  make clean"						>&2; \
-	    echo "  make -j"						>&2; \
+	    echo ""                                                     >&2; \
+	    echo "Please update your submodules like this:"             >&2; \
+	    echo "  make submods"                                       >&2; \
 	    echo ""							>&2; \
 	    exit 1; \
 	 fi
@@ -90,6 +86,10 @@ snabb: $(LUAOBJ) $(PFLUAOBJ) $(HOBJ) $(COBJ) $(ARCHOBJ) $(ASMOBJ) $(INCOBJ) $(LU
 	@ls -sh snabb
 
 all: $(EXE)
+
+# Rebuild after git submodules are updated.
+submods:
+	(cd ..; git submodule update; make clean; make)
 
 $(EXE): snabb
 	$(E) "PROGRAM   $@"

